apps/pages/src/components/EditPage.vue

```diff
@@ -1,115 +1,286 @@
+import { generateHtmlPreview } from "../utils/generateHtmlPreview";
+
+const editorLanguages = ["html", "css", "javascript"];
 
 export default {
   components: {
+    InputText,
     ButtonAction,
-    ButtonAlt,
+    ButtonOutline,
     MessageError,
     MessageSuccess,
     Spinner,
-    QuillEditor,
+    IconAction,
+    ArrayInput,
   },
   data() {
     return {
       graphqlError: null,
       success: null,
-      loading: false,
-      draft: "<h1>New page</h1><p>Add your contents here</p>",
+      loading: true,
+      html: {},
+      css: {},
+      javascript: {},
+      content: {
+        html: "",
+        css: "",
+        javascript: "",
+        dependencies: {
+          css: [],
+          javascript: [],
+        },
+      },
+      cssAssets: [],
+      jsAssets: [],
     };
   },
   props: {
     page: String,
     session: Object,
   },
   computed: {
+    pageSettingKey() {
+      return "page." + this.page;
+    },
     title() {
       if (
         this.session &&
         this.session.settings &&
         this.session.settings["page." + this.page]
-      )
-        return "Edit page '" + this.page + "'";
-      else return "Create new page '" + this.page + "'";
+      ) {
+        return "'" + this.page + "' page editor";
+      } else {
+        return "Create new page '" + this.page + "'";
+      }
+    },
+    contentJSON() {
+      return JSON.stringify(this.content);
     },
   },
   methods: {
-    savePage() {
-      this.loading = true;
-      this.graphqlError = null;
-      this.success = null;
-      request(
+    async savePageSettings() {
+      const response = await request(
         "graphql",
-        `mutation change($settings:[MolgenisSettingsInput]){change(settings:$settings){message}}`,
+        `mutation change($settings:[MolgenisSettingsInput]){change(settings:$settings){status message}}`,
         {
           settings: {
-            key: "page." + this.page,
-            value: this.draft.trim(),
+            key: this.pageSettingKey,
+            value: this.contentJSON,
           },
         }
-      )
-        .then((data) => {
-          this.success = data.change.message;
-          this.session.settings["page." + this.page] = this.draft;
-        })
-        .catch((graphqlError) => {
-          this.graphqlError = graphqlError.response.errors[0].message;
-        })
-        .finally((this.loading = false));
-    },
-    reload() {
-      if (
-        this.session &&
-        this.session.settings &&
-        this.session.settings["page." + this.page]
-      ) {
-        this.draft = this.session.settings["page." + this.page];
+      );
+      if (Object.hasOwn(response, "change")) {
+        if (response.change.status === "SUCCESS") {
+          this.success = response.change.message;
+        }
       } else {
-        return "New page, edit here";
+        this.graphqlError = response;
+      }
+    },
+    initEditor(editor) {
+      const editorRef = this.$refs[editor];
```

Comment:
if the current contents is not a json object, call a js function that will migrate.

Reply:
Added